lib/JSON/Validator.pm

```diff
@@ -529,7 +537,15 @@ sub _validate_all_of {
   my $expected = join ' or ', _uniq(@expected);
   return E $path, "allOf failed: Expected $expected, not $type."
     if $expected and @errors + @expected == @$rules;
-  return E $path, sprintf 'allOf failed: %s', _merge_errors(@errors) if @errors;
+
+  if (@errors) {
+    if ( $self->{errors_conf}->{complex_as_list} ) {
+        return _flatten(@errors);
```

Comment:
How will you know that it was an allOf error when you just return a list of errors?

Reply:
We don't. But is it important ? This is really a question for me to understand sorry. Isn't all the "...Of" just a way to group "fields" or "objects" ?

On the example of issue #116 I thought I could reference a property by '/allOfThem/prop3". Now I'm seeing it should be 'allOfThem/allOf/prop3' ? That's awfull :-(

Reply:
I think we need a complex test case to see how it plays out in real life.